File: aspnetcore/Repositories/UserRepository.cs
using local_ly_dotnet.Models;

namespace local_ly_dotnet.Repositories;

public class UserRepository
{
    private readonly List<User> _users =
    [
        new User
        {
            Id = "1",
            Name = "Gianfranco",
            Email = "gianfranco@email.com",
            Password = "Test123!",
        },
        new User
        {
            Id = "2",
            Name = "Gianfranco2",
            Email = "gianfranco@email.com",
            Password = "Test123!",
        },
        new User
        {
            Id = "3",
            Name = "Gianfranco3",
            Email = "gianfranco@email.com",
            Password = "Test123!",
        },
        new User
        {
            Id = "4",
            Name = "Gianfranco4",
            Email = "gianfranco@email.com",
            Password = "Test123!",
        },
        new User
        {
            Id = "5",
            Name = "Gianfranco5",
            Email = "gianfranco@email.com",
            Password = "Test123!",
        },
    ];

    public List<User> GetAllUsers() => _users;

    public User? GetUserById(string id) => _users.FirstOrDefault(u => u.Id == id);

    public User? GetUserByEmail(string email) => _users.FirstOrDefault(u => u.Email == email);
}

File: bun/src/repositories/user-repository.ts
import type { User } from '../models/user.js';

export class UserRepository {
  private users: User[] = [
    { id: '1', name: 'Gianfranco', email: 'gianfranco@email.com', password: 'Test123!' },
    { id: '2', name: 'Gianfranco2', email: 'gianfranco@email.com', password: 'Test123!' },
    { id: '3', name: 'Gianfranco3', email: 'gianfranco@email.com', password: 'Test123!' },
    { id: '4', name: 'Gianfranco4', email: 'gianfranco@email.com', password: 'Test123!' },
    { id: '5', name: 'Gianfranco5', email: 'gianfranco@email.com', password: 'Test123!' },
  ];

  getAllUsers(): User[] {
    return this.users;
  }

  getUserById(id: string): User | undefined {
    return this.users.find(u => u.id === id);
  }

  getUserByEmail(email: string): User | undefined {
    return this.users.find(u => u.email === email);
  }
}
